src/coreclr/jit/gentree.cpp

```diff
@@ -20533,8 +20533,9 @@ bool GenTree::isEmbeddedMaskingCompatibleHWIntrinsic() const
             }
         }
 #elif defined(TARGET_ARM64)
-        return HWIntrinsicInfo::IsEmbeddedMaskedOperation(AsHWIntrinsic()->GetHWIntrinsicId()) ||
-               HWIntrinsicInfo::IsOptionalEmbeddedMaskedOperation(AsHWIntrinsic()->GetHWIntrinsicId());
+        return HWIntrinsicInfo::IsEmbeddedMaskedOperation(intrinsicId) ||
+               HWIntrinsicInfo::IsOptionalEmbeddedMaskedOperation(intrinsicId) ||
+               (intrinsicId == NI_Sve_ConditionalSelect);
```

Comment:
should this condition instead be part of one of `IsEmbeddedMaskedOperation` or `IsOptionalEmbeddedMaskedOperation` to make sure there are no missing scenarios?

Reply:
the summar docs of this method should change and include arm64 portion as well.

Reply:
looks like `isEmbeddedMaskingCompatibleHWIntrinsic` is just used at 2 places - `canBeContained()` and `ContainCheckHWIntrinsic`. For the later, we make sure that it is not a `CndSel `, so essentially we just need this condition for `canBeContained()`. Wondering if you can add this condition somewhere like `getContainedCselOperand()` itself?

Reply:
Moved the containable check to `isContainableHWIntrinsic`. Apparently, csel is the first Arm64 one there.